src/BSON/Int64.c

```diff
@@ -244,6 +263,100 @@ static int php_phongo_int64_compare_objects(zval* o1, zval* o2)
 	return 0;
 }
 
+static zend_result php_phongo_int64_cast_object(phongo_compat_object_handler_type* readobj, zval* retval, int type)
+{
+	php_phongo_int64_t* intern;
+
+	intern = Z_OBJ_INT64(PHONGO_COMPAT_GET_OBJ(readobj));
+
+	switch (type) {
+		case IS_LONG:
+#if PHP_VERSION_ID >= 70300
+		case _IS_NUMBER:
+#endif
+#if SIZEOF_ZEND_LONG == 4
+			if (intern->integer > INT32_MAX || intern->integer < INT32_MIN) {
+				zend_error(E_WARNING, "Truncating 64-bit integer value %" PRId64 " to 32 bits", intern->integer);
+			}
+#endif
+
+			ZVAL_LONG(retval, intern->integer);
+
+			return SUCCESS;
+
+		case _IS_BOOL:
+			ZVAL_BOOL(retval, intern->integer != 0);
+
+			return SUCCESS;
+
+		default:
+			return zend_std_cast_object_tostring(readobj, retval, type);
+	}
+}
+
+static zend_result php_phongo_int64_do_operation(zend_uchar opcode, zval* result, zval* op1, zval* op2)
+{
+	int64_t value1, value2;
+
+	if (Z_TYPE_P(op1) == IS_LONG) {
+		value1 = Z_LVAL_P(op1);
+	} else if (Z_TYPE_P(op1) == IS_OBJECT && Z_OBJCE_P(op1) == php_phongo_int64_ce) {
+		value1 = Z_INT64_OBJ_P(op1)->integer;
+	} else {
+		return FAILURE;
+	}
+
+	if (Z_TYPE_P(op2) == IS_LONG) {
+		value2 = Z_LVAL_P(op2);
+	} else if (Z_TYPE_P(op2) == IS_OBJECT && Z_OBJCE_P(op2) == php_phongo_int64_ce) {
+		value2 = Z_INT64_OBJ_P(op2)->integer;
+	} else {
+		return FAILURE;
+	}
+
+	switch (opcode) {
+		case ZEND_ADD:
+			ZVAL_INT64_OBJ(result, value1 + value2);
+			return SUCCESS;
+
+		case ZEND_SUB:
+			ZVAL_INT64_OBJ(result, value1 - value2);
+			return SUCCESS;
+
+		case ZEND_MUL:
+			ZVAL_INT64_OBJ(result, value1 * value2);
+			return SUCCESS;
+
+		case ZEND_DIV:
+			if (value2 == 0) {
+				zend_throw_exception(zend_ce_division_by_zero_error, "Division by zero", 0);
+				return FAILURE;
+			}
+
+			ZVAL_INT64_OBJ(result, value1 / value2);
+			return SUCCESS;
+
+		case ZEND_MOD:
+			ZVAL_INT64_OBJ(result, value1 % value2);
+			return SUCCESS;
+
+		case ZEND_SL:
+			ZVAL_INT64_OBJ(result, value1 << value2);
+			return SUCCESS;
+
+		case ZEND_SR:
+			ZVAL_INT64_OBJ(result, value1 >> value2);
+			return SUCCESS;
+
+		case ZEND_POW:
+			ZVAL_INT64_OBJ(result, pow(value1, value2));
```

Comment:
Note that `pow()` returns a `float`, so this can easily result in silent truncation based on `ZVAL_INT64_OBJ` calling `php_phongo_bson_new_int64` and C implicitly casting the rturn value to an `int64_t`.

You can contrast this with [`pow_function` in `zend_operators.c`](https://github.com/php/php-src/blob/php-8.2.5/Zend/zend_operators.c#L1288), which has much more complex handling to decide if the result can be represented as an integer or must utilize a float.

If we intentionally omit this opcode here, we could rely on `pow_function` to convert the Int64 to a numeric type. That would make `$int64a ** $int64b` would be more likely to return a PHP float, but it'd be more predictable and consistent with how `**` might work with regular integers.

Whatever we decide, we should ensure there are tests for the overflow conditions. Also, some tests for a negative `value2` operand. Currently, I think a negative `value2` would always produce `Int64(0)` as a result.